### PR TITLE
More unit testifications

### PR DESCRIPTION
**Pull request by rbtcollins**
See the individual commits please

### CODE REVIEW

File: tests/cli-self-upd.rs [hunk 1]
```diff
@@ -494,32 +494,25 @@ fn when_cargo_home_is_the_default_write_path_specially() {
...
-
-        let path = config.cargodir.join("bin").to_string_lossy().to_string();
         assert!(!get_path().unwrap().contains(&path));
     });
 }
 
 #[test]
 #[cfg(unix)]
+/// This covers both windows and unix no-path-modification cases due to the code structure.
 fn install_doesnt_modify_path_if_passed_no_modify_path() {
```
Comment (workingjubilee) [inline tests/cli-self-upd.rs:511]: This test will not compile on Windows as it is set to be `#[cfg(unix)]`, and so provides no test coverage on Windows. This is observable: It is absent from the build logs as well.

Comment (rbtcollins) [reply tests/cli-self-upd.rs:511]: yes, I know. I think this is fine: we run windows and unix tests together in CI, so the test won't fail to run.

The code structure in the library means that if this test fails, we know that the platform specific do_add_to_path routines are being called inappropriately, and thats from the platform agnostic code - but because the routines that are being called are platform specific the only way we can detect this is by having some platform specific mechanism.

As this is a blackbox test, we can't drop a replacement trait in and mock it without compromising the binary that we build; a future unit test in self_update.rs might provide better coverage in a platform neutral way with a unit-test-specific trait object, but for now I think this is fine.

The trade off is that a windows developer that breaks this test will only find out in CI, which I think is ok.

Comment (workingjubilee) [reply tests/cli-self-upd.rs:511]: Fair enough.

Comment (rbtcollins) [reply tests/cli-self-upd.rs:511]: Oh, I should add, the equivalent platform specific test on the windows side has to modify global state - the windows registry - which is potentially disruptive, and not worth doing for an edge case test in the first place when other coverage means are available.

File: src/lib.rs [hunk 2]
```diff
@@ -66,5 +66,7 @@ pub mod fallback_settings;
 mod install;
 mod notifications;
 pub mod settings;
+// #[cfg(test)]
```
Comment (kinnison) [inline src/lib.rs]: ?

**Review by kinnison — COMMENTED** [review]
It looks consistent, modulo the -gnu failure case we need to resolve, but what's that comment all about?

File: src/lib.rs (hunk 2 above)
Comment (rbtcollins) [reply src/lib.rs]: oh hmm, I think I wanted that to be cfg(test) but it can't be for the integration tests to use the test helpers, they need to be built normally and then dropped during link

**Review by kinnison — APPROVED** [review]
This all seems to make sense.

**Comment by kinnison** [comment]
You are right that it took quite a kicking from some heavy boots, but I think this is good.
